packages/zeplin-extension-style-kit/values/gradient/radialGradient.js

```diff
@@ -0,0 +1,58 @@
+import LinearColorStop from "./linearColorStop";
+import Percent from "../percent";
+
+class RadialGradient {
+    // eslint-disable-next-line max-params
+    constructor(colorStops, from, to, width, height) {
+        this.center = from;
+
+        const x1 = from.x * width;
+        const y1 = from.y * height;
+        const x2 = to.x * width;
+        const y2 = to.y * height;
+
+        const gradientDistance = Math.hypot(x1 - x2, y1 - y2);
+
+        let farthestDistance = 0;
+        const corners = [
+            { x: 0, y: 0 },
+            { x: width, y: 0 },
+            { x: width, y: height },
+            { x: 0, y: height }
+        ];
+
+        corners.forEach(corner => {
+            const distance = Math.hypot(x1 - corner.x, y1 - corner.y);
+            if (distance > farthestDistance) {
+                farthestDistance = distance;
+            }
+        });
+
+        this.colorStops = colorStops.map(cs => new LinearColorStop(cs, 0, gradientDistance, 0, farthestDistance));
```

Comment:
The same comment about leaking calculating color stop position into its constructor applies to here too. We can calculate here and give position directly.